roseau/load_flow/utils/mixins.py

```diff
@@ -171,14 +171,9 @@ def to_json(self, path: StrPath, *, include_results: bool = True) -> Path:
             The expanded and resolved path of the written file.
         """
         res = self.to_dict(include_results=include_results)
-        output = json.dumps(res, ensure_ascii=False, indent=2, default=_json_encoder_default)
-        # Collapse multi-line arrays of 2-to-4 elements into single line
-        # e.g complex value represented as [real, imag] or rows of the z_line matrix
-        output = re.sub(r"\[(?:\s+(\S+,))?(?:\s+?( \S+,))??(?:\s+?( \S+,))??\s+?( \S+)\s+]", r"[\1\2\3\4]", output)
-        if not output.endswith("\n"):
-            output += "\n"
         path = Path(path).expanduser().resolve()
-        path.write_text(output)
+        with path.open("w", encoding="utf-8") as fp:
+            json.dump(res, fp, ensure_ascii=False, indent=2, default=_json_encoder_default)
```

Comment:
It is maybe time to use `orjson`?

Reply:
I am planning to add support for orjson later as an optional, used by default if installed, json engine.